cache/sampleStoreTimeout.js

```diff
@@ -34,61 +94,46 @@ module.exports = {
-
-        allSamples.forEach((sampKey) => {
-          const aspectName = sampKey.split('|')[ONE];
-          commands.push(['hgetall', sampKey]); // get sample
-          commands.push(
-            ['hgetall', sampleStore.toKey(aspectType, aspectName)]
-          );
-        });
-
-        return redisClient.batch(commands).execAsync();
-      })
-      .then((redisResponses) => {
-        const samples = [];
-        const aspects = [];
-        const sampCmds = [];
-
-        for (let num = 0; num < redisResponses.length; num += TWO) {
-          samples.push(redisResponses[num]);
-          aspects.push(redisResponses[num + ONE]);
-        }
+    return redisClient.smembersAsync(sampleStore.constants.indexKey.sample)
+    .then((allSamples) => {
+      const commands = [];
+      const aspectsSet = new Set();
+      const aspectType = sampleStore.constants.objectType.aspect;
 
-        for (let num = 0; num < samples.length; num++) {
-          const samp = samples[num];
-          const asp = aspects[num];
-          const sampUpdDateTime = new Date(samp.updatedAt);
-          if (asp && isTimedOut(asp.timeout, curr, sampUpdDateTime)) {
-            const sampType = sampleStore.constants.objectType.sample;
+      /**
+       * Add commands to get all samples first, create an aspect set with
+       * unique aspect names. Add commands to get the aspects then.
+       */
+      samplesCount = allSamples.length;
+      allSamples.forEach((sampKey) => {
+        const aspectName = sampKey.split('|')[ONE];
+        aspectsSet.add(aspectName);
+        commands.push(['hgetall', sampKey]);
+      });
 
-            const objToUpdate = {
-              value: constants.statuses.Timeout,
-              status: constants.statuses.Timeout,
-              previousStatus: samp.status,
-              statusChangedAt: new Date().toString(),
-              updatedAt: new Date().toString(),
-            };
-            sampCmds.push([
-              'hmset',
-              sampleStore.toKey(sampType, samp.name),
-              objToUpdate,
-            ]);
-            numberTimedOut++;
-          }
-        }
+      aspectsSet.forEach((aspName) => {
+        commands.push(
+          ['hgetall', sampleStore.toKey(aspectType, aspName)]
+        );
+      });
 
-        numberEvaluated = samples.length;
-        return redisClient.batch(sampCmds).execAsync();
-      })
-      .then(() => resolve({ numberEvaluated, numberTimedOut }))
-      .catch(reject);
+      return redisClient.batch(commands).execAsync();
+    })
+    .then((redisResponses) => {
+      const sampCmds = getSampleTimeoutCommands(
+      redisResponses, samplesCount, curr);
+      numberEvaluated = samplesCount;
+      numberTimedOut = sampCmds.length;
+      return redisClient.batch(sampCmds).execAsync();
+    })
+    .then(() => {
+      const res = { numberEvaluated, numberTimedOut };
+      return res;
+    })
+    .catch((err) => {
```

Comment:
if we're not doing anything but just throwing the same err, then we shouldn't need that catch block at all here